src/main/webapp/view/transcript/transcript.js

```diff
@@ -113,60 +70,101 @@ function seekTranscript(currentTime) {
+  currentTranscriptLine.removeBold();
   currentTranscriptLine = currentTranscriptLine.nextElementSibling;
-  scrollToTopOfTranscript(currentTranscriptLine);
-  addBold(currentTranscriptLine);
+  currentTranscriptLine.scrollToTopOfTranscript();
+  currentTranscriptLine.addBold();
   // TODO: Handle the case where the video isn't only playing.
 }
 
 /**
- * Bolds the text in `transcriptLineLiElement` if it is not already
- * bolded.
+ * Creates a transcript line element containing the text,
+ * start time, and end time.
  */
-function addBold(transcriptLineLiElement) {
-  if (isBolded(transcriptLineLiElement)) {
-    return;
+class TranscriptLine extends HTMLElement {
+  /**
+   * Creates a custom HTML element representing a transcript line.
+   *
+   * <p>Uses the template and slots defined in `TRANSCRIPT_TEMPLATE` to
+   * help create the transcript line.
+   *
+   * @param transcriptLine The transcriptLine from `ENDPOINT_TRANSCRIPT`
+   *     whose `attributes` should be used.
+   */
+  constructor(transcriptLine) {
+    super();
+    const timestampRange = window.createTimestampRange(
+        transcriptLine.startTimestampMs, transcriptLine.endTimestampMs);
+
+    const template = document.getElementById(TRANSCRIPT_TEMPLATE);
+    this.attachShadow({mode: 'open'});
+    this.shadowRoot.appendChild(template.content.cloneNode(true));
+    this.updateTemplateSlot(TRANSCRIPT_SLOT_TIME_RANGE, timestampRange);
+    this.updateTemplateSlot(TRANSCRIPT_SLOT_CONTENT, transcriptLine.content);
+
+    this.transcriptLine = transcriptLine;
   }
-  transcriptLineLiElement.classList.add(BOLD_FONT_WEIGHT);
-  transcriptLineLiElement.classList.remove(DEFAULT_FONT_WEIGHT);
-}
 
-/**
- * Removes bold from the text in `transcriptLineLiElement` if it
- * is currently bolded.
- */
-function removeBold(transcriptLineLiElement) {
-  if (!isBolded(transcriptLineLiElement)) {
-    return;
+  /**
+   * Updates the template slot `slotName` with `slotValue`.
+   */
+  updateTemplateSlot(slotName, slotValue) {
+    const span = document.createElement('span');
+    span.innerText = slotValue;
+    span.slot = slotName;
+    this.appendChild(span);
   }
-  transcriptLineLiElement.classList.add(DEFAULT_FONT_WEIGHT);
-  transcriptLineLiElement.classList.remove(BOLD_FONT_WEIGHT);
-}
 
-/** Returns true if`transcriptLineLiElement` is bolded. */
-function isBolded(transcriptLineLiElement) {
-  return transcriptLineLiElement.classList.contains(BOLD_FONT_WEIGHT);
-}
+  /** Returns true if the element is already bolded. */
+  isBolded() {
+    return this.classList.contains(BOLD_FONT_WEIGHT);
+  }
 
-/**
- * Returns true if `currentTimeMs` is within the time range for
- * the current transcript line.
- */
-function isWithinCurrentTimeRange(currentTimeMs) {
-  return currentTranscriptLine.startTimestampMs <= currentTimeMs &&
-      currentTimeMs <= currentTranscriptLine.endTimestampMs;
-}
+  /**
+   * Bolds the element if it is not already
+   * bolded.
```

Comment:
nit: the new line for the work bolded seems a bit weird. You should only add new lines if it goes above a certain character length.